var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -685,6 +687,9 @@ def cmake_args(self):
             cmake_args.append(
                 '-DLLVM_TARGETS_TO_BUILD:STRING=' + ';'.join(targets))
 
+        if spec.satisfies('@6.0.0:') and '+omp_tsan' in spec:
```

Comment:
Could you replace the `satisfies` check with a `conflicts` directive?

`conflicts('+omp_tsan', when='@:5.9.9')`

If you prefer otherwise that's fine, I think it might avoid unexpected outcomes though if someone builds with `+omp_tsan` with an older version.